### PR TITLE
Added a report tool

### PR DESCRIPTION
**Pull request by micheles**
See https://bugs.launchpad.net/oq-engine/+bug/1471648. Also, a lot of refactoring was done.
The tests are green: https://ci.openquake.org/job/zdevel_oq-risklib/836/

### CODE REVIEW

File: openquake/commonlib/calculators/views.py [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+    return '\n'.join(lines)
+
+
+def classify_gsim_lt(gsim_lt):
+    """
+    :returns: "trivial", "simple" or "complex"
+    """
+    trt_gsims = gsim_lt.values.items()
+    num_gsims = map(len, gsim_lt.values.itervalues())
+    complex_trts = [trt for trt, gsims in trt_gsims if len(gsims) > 1]
```
Comment (ptormene) [inline openquake/commonlib/calculators/views.py]: Please rename into `possibly_complex_trts`

File: openquake/commonlib/calculators/views.py [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+    if all(n == 1 for n in num_gsims):  # one gsim per TRT
```
Comment (ptormene) [inline openquake/commonlib/calculators/views.py]: Please change into `if len(multi_gsim_trts) == 0`

File: openquake/commonlib/calculators/views.py [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+
+@view.add('inputs')
+def view_inputs(token, dstore):
+    inputs = dstore['oqparam'].inputs.copy()
+    try:
+        source_models = [('source', fname) for fname in inputs['source']]
+        del inputs['source']
+    except KeyError:  # there is no 'source' in scenario calculations
+        source_models = []
+    return rst_table([['Name', 'File']] + hide_fullpath(
```
Comment (ptormene) [inline openquake/commonlib/calculators/views.py]: Please make ['Name', 'File'] a header

File: openquake/commonlib/commands/info.py [hunk 4]
```diff
@@ -116,26 +125,28 @@ def _info(name, filtersources, weightsources):
...
+         datatransfer=False, report=False):
     """
     Give information. You can pass the name of an available calculator,
     a job.ini file, or a zip archive with the input files.
     """
     logging.basicConfig(level=logging.INFO)
     with Monitor('info', measuremem=True) as mon:
-        if datatransfer:
+        if report:
+            print('Generated', reportwriter.build_report(name, '/tmp'))
```
Comment (ptormene) [inline openquake/commonlib/commands/info.py]: Please use tempdir

File: openquake/commonlib/calculators/views.py (hunk 1 above)
Comment (micheles) [inline openquake/commonlib/calculators/views.py]: Done

File: openquake/commonlib/calculators/views.py (hunk 2 above)
Comment (micheles) [inline openquake/commonlib/calculators/views.py]: Done

File: openquake/commonlib/calculators/views.py (hunk 3 above)
Comment (micheles) [inline openquake/commonlib/calculators/views.py]: Done

File: openquake/commonlib/commands/info.py (hunk 4 above)
Comment (micheles) [inline openquake/commonlib/commands/info.py]: Done

File: doc/effective-realizations.rst [hunk 5]
```diff
@@ -82,70 +129,33 @@ three different paths in the first tectonic region type will produce
...
 == ======
-0  `*_D`
-1  `*_E`
+0  `@_D`
+1  `@_E`
 == ======
 
-In such situation the engine will perform the computation only for the 2
+The "@" character should be read as "any", meaning that for the first
+tectonic region type any path (i.e. both "A", "B" and "C") will give
```
Comment (ptormene) [inline doc/effective-realizations.rst]: Please remove 'both', or change it

File: doc/effective-realizations.rst [hunk 6]
```diff
@@ -1,23 +1,69 @@
...
+optimization to take advantage of such situations. Here I will
+explain how the optimization work.
+
+First, it is best to give some terminology.
+
+1. for each source model in the source model logic tree there is a
+   different GMPE logic tree
+2. the total number of realizations is the sum of the number of realizations
+   of each GMPE logic tree
+3. a GMPE logic tree is *trivial* if has no tectonic region types with
```
Comment (ptormene) [inline doc/effective-realizations.rst]: In points 3, 4 and 5, please add 'it' (...if it has...)

File: doc/event-based-with-large-logic-tree.rst [hunk 7]
```diff
@@ -0,0 +1,255 @@
+Event based computations with complex logic trees
+-------------------------------------------------
+
+Previously_ we discussed the concept of effective realizations and
```
Comment (ptormene) [inline doc/event-based-with-large-logic-tree.rst]: Please remove the _ unless it's some kind of link to a previous section

File: doc/event-based-with-large-logic-tree.rst [hunk 8]
```diff
@@ -0,0 +1,255 @@
...
+with the ``-r`` flag, which produces a report in `.rst` format
+containing a lot of useful information about the logic tree
+and the ruptures. In the case of an event based
+calculation you should always run
+
+`$ oq-lite info -r job.ini`
+
+It will be slower than just filtering the sources but reliable.
+
+.. _Previously: effective-realizations.rst
```
Comment (ptormene) [inline doc/event-based-with-large-logic-tree.rst]: This seems to answer my previous question

File: doc/event-based-with-large-logic-tree.rst [hunk 9]
```diff
@@ -0,0 +1,255 @@
...
+source model repeated 4,000 times, because of the inherent
+stochasticity of the process. Therefore the time spent in generating
+the needed amount of ruptures could make the calculator slower than
+using full enumeration: remember than when using full enumeration the
+ruptures of a given source model are generated exactly once, since
+each path is taken exactly once.
+
+Notice that even if source model path is sampled several times, the
+model is parsed and sent to the workers *only once*. In particular if
+there is a single source model and `number_of_logic_tree_samples =
```
Comment (ptormene) [inline doc/event-based-with-large-logic-tree.rst]: Github is not reading correctly the multiline code, but I suppose it works in general. Please be sure of that.

**Comment by ptormene** [comment]
Apart from the minor comments, LGTM

File: doc/effective-realizations.rst (hunk 5 above)
Comment (micheles) [inline doc/effective-realizations.rst]: Done

File: doc/effective-realizations.rst (hunk 6 above)
Comment (micheles) [inline doc/effective-realizations.rst]: Done

**Pull request merged**